src/examples/learn_quant/monotonicity.py

```diff
@@ -0,0 +1,254 @@
+from ultk.effcomm.informativity import informativity
+from ultk.language.grammar import GrammaticalExpression
+from ultk.language.language import Language, aggregate_expression_complexity
+from ultk.language.semantics import Meaning
+
+from learn_quant.meaning import create_universe
+from learn_quant.util import read_expressions
+from learn_quant.quantifier import QuantifierUniverse, QuantifierModel
+from concepts.contexts import Context
+
+from itertools import product
+import math
+from scipy.stats import entropy
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+from scipy import sparse
+
+from pathlib import Path
+
+import hydra
+from omegaconf import DictConfig, OmegaConf
+
+
+def switch_direction(bools: np.ndarray, down=False) -> np.ndarray:
+    """Switches the value of True to False and vice versa.
+
+    Args:
+        bool (np.ndarray): Accepts a list of lists of bools
+
+    Returns:
+        np.ndarray: A list of lists of bools, flipped unless down == True.
+    """
+    if down:
+        return bools
+    else:
+        return bools
+
+
+def get_truth_matrix(universe: QuantifierUniverse) -> np.ndarray:
+    """Get a numpy matrix that stores the truth vectors of all models in the quantifier universe
+
+    Returns:
+        np.ndarray: A matrix of booleans arrays that track the indices of set items of A intersected by B.
+    """
+    truth_array = []
+    names_array = []
+    for quantifier_model in universe.referents:
+        truth_vector = tuple(True if x == "2" else False for x in quantifier_model.name)
+        truth_array.append(truth_vector)
+        names_array.append(quantifier_model.name)
+    truth_matrix = np.array(truth_array)
+    names_vector = np.array(names_array)
+    return truth_matrix, names_vector
```

Comment:
There's a mismatch here between type annotation and return type (which is `tuple[np.ndarray, np.ndarray]`).  I'd recommend setting up VSCode with `mypy` so that it automatically alerts you to these things and they can be spotted quickly :).  Also: why does names need to be an array?

Reply:
Do you recommend one mypy extension over another?
<img width="472" alt="image" src="https://github.com/CLMBRs/ultk/assets/147190787/4e2c5d53-f13e-4295-bcf9-b35b7582a915">

Reply:
I use the MS one, but don't have any strong/antecedent views here (probably chose it just b/c VSCode is also an MS product, honestly don't remember)